src/wallet/wallet.cpp

```diff
@@ -4713,6 +4713,31 @@ void CWallet::GetScriptForMining(std::shared_ptr<CReserveScript> &script)
     script->reserveScript = CScript() << ToByteVector(pubkey) << OP_CHECKSIG;
 }
 
+void CWallet::GetScriptForMining(std::shared_ptr<CReserveScript> &script)
+{
+    std::string sAddress = gArgs.GetArg("-miningaddress", "");
+    if (!sAddress.empty()) {
+        CTxDestination dest = DecodeDestination(sAddress);
+
+        if (IsValidDestination(dest)) {
+            if (IsMine(dest) != ISMINE_SPENDABLE) {
+                WalletLogPrintf("-miningaddress %s not under my control, falling back to random address", sAddress);
+                return GetScriptForMiningReserveKey(script);
+            }
```

Comment:
These are good checks, but they should be done where miningaddress is being checked on startup.

Reply:
Had to move it to init.cpp so I could be sure the wallet was loaded, but it works for both veild and veil-qt (with a delay to load the wallet, of course). Done.

Reply:
Ahh yes, because of the IsMine().  Let me think about this a little bit more.  I put in a good address that I didn't own, and I didn't see an error anywhere.  I'm building this PR so I can do some extra debugging later.